File: leet-sync/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int[] arr={0,0};
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                arr[0]=i;
                arr[1]=map.get(target-nums[i]);
                return arr;
            }
            map.put(nums[i],i);
        }
        return arr;
        
    }
}

File: leet-sync/1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
   m := make(map[int]int) 
   for i,val := range nums{
        valinmap,ok := m[target-val]
        if !ok{
            m[val]=i
        } else{
            return []int{valinmap,i}
        }
     
   }
   return []int{}
}